caffe2/core/IValueInterface.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <ATen/core/ivalue_to.h>
+#include <c10/util/ArrayRef.h>
+#include <caffe2/core/tensor.h>
+
+
+namespace c10 {
+struct IValue;
+}
+
+namespace caffe2 {
+namespace detail {
+
+TORCH_API caffe2::Tensor contiguous(caffe2::Tensor);
+
+class TORCH_API IValueInterface {
+public:
+  IValueInterface(): values_(nullptr), size_(0) {}
+  IValueInterface(c10::ArrayRef<c10::IValue> values);
+  ~IValueInterface();
+
+  bool isTensorList(size_t idx) const;
+  bool isTensor(size_t idx) const;
+
+  size_t size() const {
+    return size_;
+  }
+  bool empty() const {
+    return size() == 0;
+  }
+
+  const c10::IValue& at(size_t idx) const;
+  std::vector<caffe2::Tensor> toTensorVector(size_t idx) const;
```

Comment:
please document the preconditions of these functions, e.g.
`// Expects(isTensorList(idx))` for `toTensorVector`, presumably